src/main/java/bio/terra/service/filedata/azure/util/BlobCrl.java

```diff
@@ -128,6 +131,29 @@ public boolean deleteBlob(String blobName) {
     }
   }
 
+  public boolean deleteBlobsWithPrefix(String blobName) {
+    final String prefix;
+    if (blobName.endsWith("/")) {
+      prefix = blobName.substring(0, blobName.length() - 1);
+    } else {
+      prefix = blobName;
+    }
+    BlobContainerClient blobContainerClient = blobContainerClientFactory.getBlobContainerClient();
+    List<BlobItem> blobsToDelete =
+        blobContainerClient.listBlobs().stream()
```

Comment:
I'm not sure for listBlobs...I think you might want to have that stream in a try with resources since it's a lazy iterable and I think needs to be closed